pymodbus/client/base.py

```diff
@@ -210,15 +210,15 @@ async def async_execute(self, request=None):
 
         return resp
 
-    def callback_data(self, data: bytes, addr: tuple = None) -> int:
+    def callback_data(self, data: bytes, addr: tuple | None = None) -> int:
         """Handle received data
 
         returns number of bytes consumed
         """
         self.framer.processIncomingPacket(data, self._handle_response, slave=0)
         return len(data)
 
-    def callback_disconnected(self, _reason: Exception) -> None:
+    def callback_disconnected(self, _reason: Exception | None) -> None:
```

Comment:
Can that really be None ? where is it called with None ?

Reply:
See my comment 2

Reply:
Again I review your code...All call I can see have an exception.

Reply:
In `transport/transport.py:transport_close` there is a `value.callback_disconnected(None)` as I wrote in the PR comment.